edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent/Metric.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public class Metric
+    {
+        public DateTime TimeGeneratedUtc { get; }
+        public string Name { get; }
+        public double Value { get; }
+        public string Tags { get; }
+
+        public Metric(DateTime timeGeneratedUtc, string name, double value, string tags)
+        {
+            this.TimeGeneratedUtc = Preconditions.CheckNotNull(timeGeneratedUtc, nameof(timeGeneratedUtc));
```

Comment:
if you want to enforce UTC, you should check datetime.kind should be equal to DateTimeKind.Utc; otherwise throw exception.

Reply:
And you can't pass null for DateTime type